src/app/components/modal/modal.component.ts

```diff
@@ -98,21 +100,23 @@ export class ModalComponent extends BaseComponent implements OnInit {
         this.componentRef = this.container.createComponent(factory);
 
         const instance = this.componentRef.instance as DynamicComponent;
-        instance.context = Object.assign(command.context, {
+        instance.context = Object.assign(command.context || {}, {
             _close: this._close.bind(this),
         });
 
         this._handleKBD(command.type);
     }
 
     _getComponentType(typeName: string) {
```

Comment:
I think it is not okay to accept any string here. Introduce a key-union type instead:

`export type ModalComponentType = "blank" | "dialog" | "image" | "spinner" | "filter";`

or maybe some trick with `keyof`

Reply:
Fixed